src/core/public/chrome/global_search/global_search_service.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ReactNode } from 'react';
+
+export enum SearchObjectTypes {
+  PAGES = 'pages',
+  SAVED_OBJECTS = 'saved_objects',
+}
+/**
+ * @experimental
+ */
+export interface GlobalSearchStrategy {
```

Comment:
Can we not use search strategy as the name? We already have a concept of search strategy in the application. This is used for the actual search queries. Lets call these commands. Here is a really good extensible model that VS code uses for their global search. https://code.visualstudio.com/api/references/contribution-points#contributes.commands

Reply:
rename it to searchCommand